diff --git a/dist/node/chunk-3d7P_gGE.js b/dist/node/chunk-3d7P_gGE.js
index 2970286cac137f7c15d531954cca600407fdbedd..9b7624e1152bdcb48c62c654c7e7d3861dfe4258 100644
--- a/dist/node/chunk-3d7P_gGE.js
+++ b/dist/node/chunk-3d7P_gGE.js
@@ -36951,32 +36951,41 @@ const markers = [
     end: /^\s*\(\*\s*#endregion\b\s*(.*?)\s*\*\)/
   }
 ];
-function findRegion(lines, regionName) {
-  let chosen = null;
-  for (let i = 0; i < lines.length; i++) {
-    for (const re of markers) {
-      if (re.start.exec(lines[i])?.[1] === regionName) {
-        chosen = { re, start: i + 1 };
-        break;
+function findRegions(lines, regionName) {
+  const returned = [];
+  for (const re of markers) {
+    let nestedCounter = 0;
+    let start = null;
+    for (let i = 0; i < lines.length; i++) {
+      const startMatch = re.start.exec(lines[i]);
+      if (startMatch?.[1] === regionName) {
+        if (nestedCounter === 0) start = i + 1;
+        nestedCounter++;
+        continue;
+      }
+      if (nestedCounter === 0) continue;
+      const endMatch = re.end.exec(lines[i]);
+      if (endMatch?.[1] === regionName || endMatch?.[1] === "") {
+        nestedCounter--;
+        if (nestedCounter === 0 && start != null) {
+          returned.push({ re, start, end: i });
+          start = null;
+        }
       }
     }
-    if (chosen) break;
-  }
-  if (!chosen) return null;
-  let counter = 1;
-  for (let i = chosen.start; i < lines.length; i++) {
-    if (chosen.re.start.exec(lines[i])?.[1] === regionName) {
-      counter++;
-      continue;
-    }
-    const endRegion = chosen.re.end.exec(lines[i])?.[1];
-    if (endRegion === regionName || endRegion === "") {
-      if (--counter === 0) return { ...chosen, end: i };
-    }
+    if (returned.length > 0) break;
   }
-  return null;
+  return returned;
 }
 const snippetPlugin = (md, srcDir) => {
+  function stripMarkers(lines) {
+    return lines.filter((l) => {
+      for (const m of markers) {
+        if (m.start.test(l) || m.end.test(l)) return false;
+      }
+      return true;
+    }).join("\n");
+  }
   const parser = (state, startLine, endLine, silent) => {
     const CH = "<".charCodeAt(0);
     const pos = state.bMarks[startLine] + state.tShift[startLine];
@@ -37009,26 +37018,40 @@ const snippetPlugin = (md, srcDir) => {
   md.renderer.rules.fence = (...args) => {
     const [tokens, idx, , { includes }] = args;
     const token = tokens[idx];
-    const [src, regionName] = token.src ?? [];
+    const [src, region] = token.src ?? [];
     if (!src) return fence(...args);
     if (includes) {
       includes.push(src);
     }
-    const isAFile = fs.statSync(src).isFile();
-    if (!fs.existsSync(src) || !isAFile) {
-      token.content = isAFile ? `Code snippet path not found: ${src}` : `Invalid code snippet option`;
+    if (!fs.existsSync(src)) {
+      token.content = `Code snippet path not found: ${src}`;
+      token.info = "";
+      return fence(...args);
+    }
+    if (!fs.statSync(src).isFile()) {
+      token.content = `Invalid code snippet option`;
       token.info = "";
       return fence(...args);
     }
     let content = fs.readFileSync(src, "utf8").replace(/\r\n/g, "\n");
-    if (regionName) {
+    if (region) {
       const lines = content.split("\n");
-      const region = findRegion(lines, regionName);
-      if (region) {
+      const regions = findRegions(lines, region);
+      if (regions.length > 0) {
         content = dedent(
-          lines.slice(region.start, region.end).filter((l) => !(region.re.start.test(l) || region.re.end.test(l))).join("\n")
+          stripMarkers(
+            regions.flatMap(
+              (r) => lines.slice(r.start, r.end).filter((l) => !(r.re.start.test(l) || r.re.end.test(l)))
+            )
+          )
         );
+      } else {
+        token.content = `No region #${region} found in path: ${src}`;
+        token.info = "";
+        return fence(...args);
       }
+    } else {
+      content = stripMarkers(content.split("\n"));
     }
     token.content = content;
     return fence(...args);
@@ -37918,8 +37941,8 @@ function processIncludes(md, srcDir, src, file, includes, cleanUrls) {
       if (region) {
         const [regionName] = region;
         const lines = content.split(/\r?\n/);
-        let { start, end } = findRegion(lines, regionName.slice(1)) ?? {};
-        if (start === void 0) {
+        let regions = findRegions(lines, regionName.slice(1)) ?? {};
+        if (regions.length === 0) {
           const tokens = md.parse(content, {
             path: includePath,
             relativePath: slash(path$1.relative(srcDir, includePath)),
@@ -37930,17 +37953,23 @@ function processIncludes(md, srcDir, src, file, includes, cleanUrls) {
           );
           const token = tokens[idx];
           if (token) {
-            start = token.map[1];
+            const start = token.map[1];
             const level = parseInt(token.tag.slice(1));
+            let end = void 0;
             for (let i = idx + 1; i < tokens.length; i++) {
               if (parseInt(tokens[i].tag.slice(1)) <= level) {
                 end = tokens[i].map[0];
                 break;
               }
             }
+            regions.push({ start, end });
           }
         }
-        content = lines.slice(start, end).join("\n");
+        if (regions.length > 0) {
+          content = regions.flatMap((region2) => lines.slice(region2.start, region2.end)).join("\n");
+        } else {
+          content = `No region or heading #${region} found in path: ${includePath}`;
+        }
       }
       if (range) {
         const [, startLine, endLine] = range;
